CMake/Findc-ares.cmake

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+find_package(c-ares CONFIG)
```

Comment:
It is needed by gRPC because we set `gRPC_CARES_PROVIDER` as package and cares will not be compiled during the compilation of gRPC and we need to provide one.
https://github.com/facebookincubator/velox/blob/cf5aa68b97c83d18cd436f26fdaa40f815b60117/CMake/resolve_dependency_modules/grpc.cmake#L42-L44

Without cares, I notice below error:

```
CMake Error at /usr/local/lib/python3.8/dist-packages/cmake/data/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find c-ares (missing: C_ARES_LIBRARY) (Required is at least
  version "1.13.0")
Call Stack (most recent call first):
  /usr/local/lib/python3.8/dist-packages/cmake/data/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  CMake/Findc-ares.cmake:29 (find_package_handle_standard_args)
  _build/release/_deps/grpc-src/cmake/cares.cmake:42 (find_package)
  _build/release/_deps/grpc-src/CMakeLists.txt:320 (include)
```

Reply:
Thanks can you add a comment about this.

Reply:
Added below comments to clarify. Thanks.
```
  # 'gRPC_CARES_PROVIDER' is set as 'package', which means c-ares library needs
  # to be installed on the system, instead of being built by gRPC.
  set_source(c-ares)
  resolve_dependency(c-ares)
```